### PR TITLE
Add FV support for Dirac Kernels

### LINKED ISSUE #16033: Dirac kernel with FV
## Bug Description
Using a Dirac kernel with FV variables doesn't work.

## Steps to Reproduce
Add a `DiracKernel` to a FV test and run to get `_variable is null in dirac_kernel. Are you using a finite volume variable?`

## Impact
Let us use a Dirac Kernel in a FV model.

### PR DESCRIPTION
**Pull request by cpgr**
Let me know if I got this correct @lindsayad 

Closes #16033

### CODE REVIEW

**Review by lindsayad — CHANGES_REQUESTED** [review]
Looks good @cpgr . Just the novel for you to read (and change to `NEWTON`) and the requirement change.

File: test/tests/dirackernels/constant_point_source/1d_point_source_fv.i [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  [u]
+    order = CONSTANT
+    family = MONOMIAL
+    fv = true
+  []
+  [v]
+    order = CONSTANT
+    family = MONOMIAL
+    fv = true
+  []
```
[the remark this answers is not in the document]
Comment (cpgr) [reply test/tests/dirackernels/constant_point_source/1d_point_source_fv.i:27]: No, it isn't. I just copied the existing test here and changed to FV variables. I'll remove this.

File: test/tests/dirackernels/constant_point_source/1d_point_source_fv.i [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    type = FVDirichletBC
+    variable = u
+    boundary = right
+    value = 1
+  []
+[]
+
+[Executioner]
+  type = Steady
+  solve_type = 'PJFNK'
```
[the remark this answers is not in the document]
Comment (cpgr) [reply test/tests/dirackernels/constant_point_source/1d_point_source_fv.i:82]: Makes sense. Thanks for letting me know.

**Review by lindsayad — APPROVED** [review]
Awesome, thanks @cpgr !

**Comment by lindsayad** [comment]
This is failing with distributed mesh with 5 procs (and only 5 procs). Weird. I guess I should probably look into this

**Comment by lindsayad** [comment]
@cpgr do you think you've ever run any Dirac kernel simulations in which your point source is actually on a processor boundary? That is the circumstance when we have 5 procs here. I'm thinking this failure is probably  not related to the fact that is FV...thinking this might also cause issues for a pure FE simulation

**Comment by cpgr** [comment]
Not intentionally. Was this just a case of bad luck in this test? And why isn't the equivalent FE test failing too for the same reason? I think the point source is in the same spot (I just copied this test and changed to FV variables and kernels).

**Comment by lindsayad** [comment]
Good point...

**Comment by lindsayad** [comment]
Ok, so this is a discontinuous discretization problem. Consider this code:
```c++
const Elem *
DiracKernelInfo::findPoint(Point p, const MooseMesh & mesh)
{
  // If the PointLocator has never been created, do so now.  NOTE - WE
  // CAN'T DO THIS if findPoint() is only called on some processors,
  // PointLocatorBase::build() is a 'parallel_only' method!
  if (_point_locator.get() == NULL)
  {
    _point_locator = PointLocatorBase::build(TREE_LOCAL_ELEMENTS, mesh);
    _point_locator->enable_out_of_mesh_mode();
  }

  // Check that the PointLocator is ready to start locating points.
  // So far I do not have any tests that trip this...
  if (_point_locator->initialized() == false)
    mooseError("Error, PointLocator is not initialized!");

  // Note: The PointLocator object returns NULL when the Point is not
  // found within the Mesh.  This is not considered to be an error as
  // far as the DiracKernels are concerned: sometimes the Mesh moves
  // out from the Dirac point entirely and in that case the Point just
  // gets "deactivated".
  const Elem * elem = (*_point_locator)(p);

  // The processors may not agree on which Elem the point is in.  This
  // can happen if a Dirac point lies on the processor boundary, and
  // two or more neighboring processors think the point is in the Elem
  // on *their* side.
  dof_id_type elem_id = elem ? elem->id() : DofObject::invalid_id;

  // We are going to let the element with the smallest ID "win", all other
  // procs will return NULL.
  dof_id_type min_elem_id = elem_id;
  mesh.comm().min(min_elem_id);

  return min_elem_id == elem_id ? elem : NULL;
}
```
Note that we let the lowest element ID determine which element gets to apply the impulse from the Dirac point. Here's our global element numbering for a replicated mesh for your test:
replicated
and then here is with a distributed mesh with 5 processors
distributed-5-procs
So in replicated mesh mode, your point source at 0.2 will be applied to the element with bounds [0.1, 0.2]; however, for the distributed 5-proc case, your point source at 0.2 will be applied to the element with bounds [0.2, 0.3]. For a continuous finite element discretization it makes no difference what element the point source will be applied to. But for a discontinuous formulation, one element and its elemental dof gets the whole source. This understandably changes the solution

**Pull request merged**